samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m

```diff
@@ -0,0 +1,884 @@
+
+    // Modify the internal text store.
+    [self _updateTextWithRange:range text:e.text];
+
+    // You can set the proper font or direction for the updated text based on the language by checking
+    // e.inputLanguage.  We will not do that in this sample.
+
+    // Modify the current selection.
+    newSelection.endCaretPosition = newSelection.startCaretPosition;
+
+    // Update the selection of the edit context. There is no need to notify the system
+    // because the system itself changed the selection.
+    [self _setSelection:newSelection];
+}
+
+- (void)_editContextSelectionUpdating:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextSelectionUpdatingEventArgs*)e {
+    // Set the new selection to the value specified by the system.
+    WUTCCoreTextRange* range = e.selection;
+
+    // Update the selection of the edit context. There is no need to notify the system
+    // because the system itself changed the selection.
+    [self _setSelection:range];
+}
+
+// The bounds returned drive the IME position and scrolling as to not block the
+// field with the on-screen keyboard.
+- (void)_editContextLayoutRequested:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextLayoutRequestedEventArgs*)e {
+    WUTCCoreTextLayoutRequest* request = e.request;
+    WFRect* rect = [WFRect new];
+    rect.x = 0;
+    rect.y = 0;
+    rect.width = _xamlControl.width;
+    rect.height = _xamlControl.height;
+    request.layoutBounds.controlBounds = rect;
+}
+
+// The following code specifies how you would apply any formatting to the specified range of text
+// For this sample, we do not make any changes to the format.
+// Formatting includes foreground, underlines and background colors, for IME and auto correction.
+- (void)_editContextFormatUpdating:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextFormatUpdatingEventArgs*)e {
+    // Apply text color if specified.
+    // A null value indicates that the default should be used.
+    if (e.textColor != nil) {
+        // Set text color
+    } else {
+        // Reset text to default color
+    }
+
+    // Apply background color if specified.
+    // A null value indicates that the default should be used.
+    if (e.backgroundColor != nil) {
+        // Set background color
+    } else {
+        // Reset background to default color
+    }
+
+    // Apply underline if specified.
+    // A null value indicates that the default should be used.
+    if (e.underlineType != nil) {
+        // Add underline adornment
+    } else {
+        // Remove underline adornments
+    }
+}
+
+// This indicates that an IME has started composition.  If there is no handler for this event,
+// then composition will not start.
+- (void)_editContextCompositionStarted:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextCompositionStartedEventArgs*)e {
+}
+
+- (void)_editContextCompositionCompleted:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextCompositionCompletedEventArgs*)e {
+}
+
+- (void)dealloc {
```

Comment:
Is it just me but I'd love to see the dealloc close to the init so I can compare the init/cleanup steps.